app/controllers/subscription_controller.rb

```diff
@@ -170,8 +171,8 @@ def multiple_add
         # user or node subscription
       end
     else
-      flash[:warning] = "You must be logged in to subscribe for email updates."
-      redirect_to "/login"
+      flash[:warning] = "You must be logged in to subscribe for email updates!"
+      return_to "/login"
```

Comment:
Here I was refering to something else.
I was refering to redirect_to "/login/return_to=....." type thing. Please check the other files for similar code.

Reply:
```suggestion
      return_to "/login?return_to=" + request.fullpath
```